src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part1/index.md

```diff
@@ -0,0 +1,430 @@
+
+| Service Type | Reservation | Weight | Limit |
+| :---: | :---: | :---: | :---: |
+| client | 40% | 1 | 100% |
+| background recovery | 40% | 1 | 150% |
+| background best effort | 20% | 1 | 50% |
+
+## mClock Configuration Parameters
+
+The mClock-related configuration options are set transparently when any built-in profile is selected. For more information on what configuration options are overridden in the profiles, see the section titled [mClock Built-in Profiles](https://docs.ceph.com/en/latest/rados/configuration/mclock-config-ref/#mclock-built-in-profiles).
+
+## Other Ceph Configuration Parameters
+
+It is important to note the values for the following Ceph recovery-related options overridden by the mClock profiles:
+
+- **osd_max_backfills** = 1000
+- **osd_recovery_max_active** = 1000
+
+The above options set a high limit on the number of concurrent local and remote backfill/recovery operations per OSD. Under these conditions, the capability of the mClock scheduler was tested, and the results are presented below.
+
+The Ceph configuration options related to snaptrim that were left unchanged are shown below:
+
+- **osd_pg_max_concurrent_snap_trims** = 2
+- **osd_snap_trim_cost** = 1048576
+
+## Client Ops with Background Snaptrim Ops
+
+### Test Steps
+
+1. Bring up the Ceph cluster with 4 osds.
+2. Disable pg autoscaler and scrubs.
+3. Create a client RBD pool with replication factor 3 and prefill some objects in it.
+4. Create a RADOS snap pool with replication factor 3.
+5. Prefill the snap pool with 100K 4MiB sized objects.
+6. Create a RADOS pool snapshot of the snap pool.
+7. Overwrite 80% of the objects in the snap pool. This creates 80K clone objects.
+8. Initiate I/O on the client pool for 300 secs using fio
+9. After client I/Os have stabilized for 30 secs, remove the snapshot created in step 6. This triggers the removal of the 80K clone objects as part of the snaptrim operation.
+10. During step 9 above, statistics related to the client latency and throughput are captured. The test also captures the time taken to trim the 80K objects.
+
+To summarize, the steps above create 2 pools during the test. Snaptrim is triggered on one pool, and client I/O is triggered simultaneously on the other. Statistics captured during the tests are discussed below.
+
+### Test Results
+
+#### Client Throughput Comparison
+
+The chart below shows the average client throughput comparison across the schedulers and their respective configurations. The average baseline throughput with both the schedulers was determined to be just over 23K IOPS (4KiB randrw) using fio on the client RBD pool.
+
+
+[![](images/Img1_Throughput_Across_mClock_Profiles_vs_WPQ.png)](images/Img1_Throughput_Across_mClock_Profiles_vs_WPQ.png)
+
+
+<center> Fig. 1: WPQ vs mClock Profiles - Average Client Throughput Comparison With Snaptrim Ops </center>
```

Comment:
This caption is right on top of the next paragraph. Maybe we can add a newline after it, or move it closer to the image?

Reply:
Fixed by adding line breaks. Hopefully it should look better now.